group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupConfig.java

```diff
@@ -182,6 +200,15 @@ private static void validateValues(Map<?, ?> valueMaps, GroupCoordinatorConfig g
             throw new InvalidConfigurationException(SHARE_SESSION_TIMEOUT_MS_CONFIG + " must be greater than " +
                 SHARE_HEARTBEAT_INTERVAL_MS_CONFIG);
         }
+        if (ShareGroupAutoOffsetReset.valueOf(shareAutoOffsetReset.toUpperCase(Locale.ROOT)) == ShareGroupAutoOffsetReset.NONE) {
+            throw new InvalidConfigurationException(SHARE_AUTO_OFFSET_RESET_CONFIG + " must be " +
+                ShareGroupAutoOffsetReset.LATEST + " or " + ShareGroupAutoOffsetReset.EARLIEST);
+        }
+        if (ShareGroupAutoOffsetReset.valueOf(shareAutoOffsetReset.toUpperCase(Locale.ROOT)) != ShareGroupAutoOffsetReset.LATEST &&
+            ShareGroupAutoOffsetReset.valueOf(shareAutoOffsetReset.toUpperCase(Locale.ROOT)) != ShareGroupAutoOffsetReset.EARLIEST) {
+            throw new InvalidConfigurationException(SHARE_AUTO_OFFSET_RESET_CONFIG + " must be " +
+                ShareGroupAutoOffsetReset.LATEST + " or " + ShareGroupAutoOffsetReset.EARLIEST);
+        }
```

Comment:
Do you need this check, as the config is backed by an enum then will it not be an inherent check?

Reply:
Thanks for the review. Yes you are right, I will remove this in the next commit